File: Sorting Algorithms/Insertion Sort/InsertionSort.java
class InsertionSort{


    public static void main(String[] args){
        int arr[] = {2, 1, 5, 8, 3, 9, 4, 7, 6, 0};
        int n = arr.length;

        System.out.println("Insertion Sort Algorithm using Java");

        System.out.print("\n\nArray before sorting: [ ");
        for(int i=0; i<n-1; i++){
            System.out.print(arr[i] + ", ");
        }
        System.out.println(arr[n-1]+ " ]");

        insertionSort(arr, n);

        System.out.print("\n\nArray after sorting: [ ");
        for(int i=0; i<n-1; i++){
            System.out.print(arr[i] + ", ");
        }
        System.out.println(arr[n-1]+ " ]");
    }


    public static void insertionSort(int[] arr, int n){
        int key, j;
        for(int i=1; i<n; i++){
            key = arr[i];
            j = i-1;
            for(; j>=0 && arr[j]>key; j--){
                arr[j+1] = arr[j];
            }
            arr[j+1] = key;
        }
    }
    

}

File: Algorithms/Sorting Algorithms/Insertion Sort/insertion_sort.py
def main(): 
    print("Insertion Sort Algorithm using python")

    arr: list[int] = [2, 1, 5, 8, 3, 9, 4, 7, 6, 0]

    print("\nArray before sorting: ",arr)
    insertion_sort(arr)
    print("\nArray after sorting: ",arr)


def insertion_sort(arr: list[int]) -> None:
    """Sorts a list of integers by finding correct position of next int and swapping."""
    n: int = len(arr)
    for i in range(1,n):
        key: int = arr[i]
        j: int = i-1
        while j>=0 and arr[j]>key:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    return 


if __name__ == "__main__":
    main()